vizro-core/tests/unit/vizro/actions/test_set_control.py

```diff
@@ -0,0 +1,198 @@
+import re
+
+import pytest
+
+import vizro.models as vm
+from vizro import Vizro
+from vizro.actions import set_control
+from vizro.managers import model_manager
+
+
+@pytest.fixture
+def managers_two_pages_for_set_control(standard_px_chart, standard_ag_grid):
+    """Instantiates the model_manager and the data_manager with two pages."""
+    vm.Page(
+        id="test-page-1",
+        title="test-page-1",
+        components=[
+            vm.Graph(id="scatter_chart_1", figure=standard_px_chart),
+            vm.Button(id="button_1"),
+        ],
+        controls=[
+            vm.Filter(
+                id="filter_page_1",
+                column="continent",
+            ),
+        ],
+    )
+
+    vm.Page(
+        id="test-page-2",
+        title="test-page-2",
+        components=[
+            vm.Graph(id="scatter_chart_2", figure=standard_px_chart),
+        ],
+        controls=[
+            vm.Filter(
+                id="filter_page_2_show_in_url_false",
+                column="continent",
+            ),
+            vm.Filter(
+                id="filter_page_2_show_in_url_true",
+                column="continent",
+                show_in_url=True,
+            ),
+        ],
+    )
+
+    Vizro._pre_build()
+
+
+@pytest.mark.usefixtures("managers_two_pages_for_set_control")
+@pytest.mark.filterwarnings("ignore:.*experimental.*:FutureWarning")
```

Comment:
This would be removed here and where else it's done if we move it to a global setting.